### DISCUSSION TITLE
Move all property defaults to the AppConstants (also for other dtap stages)

### DISCUSSION THREAD

**Discussion by philipsens** [frankframework/frankframework#7061] [Ideas]
Sometimes it is hard to find the default value of a property, like when it gets declared in the Java code.

Also, there is no overview of which properties will change when changing the DTAP-stage.
For example, the Ladybug is disabled when the DTAP-stage is `PRD`, but this can only be found in the code itself, where an if-statement is defined.
It would be nice if an `AppContantsPrd` would exist that overrides the default values in the `AppConstants`, like how it is done with the `DeploymentSpecifics`.

**Comment by nielsm5** [comment]
That's what StageSpecifics are for, the AppConstants are the application defaults, which indeed may be different depending on the stage. The StageSpecifics_prd.properties contains all the properties specific to the stage (in my example `prd`) you've specified.

**Comment by philipsens** [comment]
> the AppConstants are the application defaults, which indeed may be different depending on the stage

Yes, that is my point. 

The testtool has an if-statement in the code to check if it needs to be enabled. I'd like that sort of stuff to be moved to the StageSpecifics. 

And the AppConstants also has a lot of empty values or the default is located in the code e.g:
```java
@Value("${iaf-api.enabled:true}")
// or
private static final String etagCacheType = appConstants.getProperty("etag.cache.type", "ehcache");
```

**Reply by nielsm5** [reply]
Ah cool, glad we're agreeing then. Related issues: #2116 and #4847.

**Comment by philipsens** [comment]
Discussed verbally, this was not possible in the past and has not yet been fully adopted into the workflow. The AppConstants are also no longer necessary as the deploymentspecifics/stagespecifics can now be used with inheritance.

**Reply by nielsm5** [reply]
Yes, though I wouldn't say `AppConstants.properties` are no longer necessary, it would be great for all properties to be consolidated into 1 file with proper documentation. Specific overrides can be in the `StageSpecifics_zyx.properties`, so they will need proper documentation in the `AppConstants.properties` that there are different defaults depending on the stage.
